### DISCUSSION TITLE
Load Masks from Folder.

### DISCUSSION THREAD

**Discussion by BlastedRemnants** [Mikubill/sd-webui-controlnet#215] [Ideas]
I use the Dynamic Prompts extension a lot and really appreciate the variety it gives when I'm making images, and I got thinking that it'd be pretty cool if we could have something similar in Control Net. I have no idea how hard it would be to code it, but I was thinking that instead of just dropping a source picture in we could choose instead to specify a folder with images, and then at run time it just picks an image from the folder and uses that for the masking. Or maybe some options dealing with how the image gets picked, probably not everyone wants random images, but maybe there'd be a way to work in some sort of sequential loading too, would be great for making gifs and videos. Anyway just a thought, could be too hard to do but it sounds like a cool feature to me  :)

**Comment by sbuldan** [comment]
This would be also amazing for adding depth maps from 3d packages like zbrush.

**Comment by kabachuha** [comment]
If you're interested, check out Deforum's pending implementation of ControlNet (it uses this extension and then builds a compat layer on top) https://github.com/deforum-art/deforum-for-automatic1111-webui/pull/361. It makes just this, grabs an input video, unpacks it into an images folder and then uses them as a ControlNet base for each frame

**Reply by BlastedRemnants** [reply]
Thanks, I'll certainly look into that for the gifs and vids side of things!
